src/Microsoft.PowerShell.ConsoleHost/host/msh/PendingProgress.cs

```diff
@@ -582,6 +591,13 @@ internal override
             }
 
             ArrayList result = new ArrayList();
+
+            if (ExperimentalFeature.IsEnabled("PSAnsiProgress") && PSStyle.Instance.Progress.View == ProgressView.Minimal)
+            {
+                RenderHelper(result, _topLevelNodes, 0, maxWidth, rawUI);
+                return (string[])result.ToArray(typeof(string));
```

Comment:
Seems like this might be a good time to convert that `ArrayList` to a `List<string>`

Reply:
I took a look at this, but believe it's out of scope for this PR.  The progress rendering code is built around ArrayList.  There's even a comment about it at the top of PendingProgress.cs about potentially using List<string>.  I would suggest waiting until Dongbo has implemented the ability to easily measure benchmark.net before tackling such changes.